### ISSUE TITLE
Session lifetime is too short

### ISSUE THREAD

**Issue by morozov** [mnapoli/externals#19]
When using [externals.io](http://externals.io/), I get logged out in about an hour after I logged in. It looks like GitHub authentication data is stored in the PHP session which expires pretty soon:
```
↪  curl -vs http://externals.io/ -o /dev/null 
> GET / HTTP/1.1
> Host: externals.io
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Date: Mon, 09 Jan 2017 19:38:36 GMT
< Content-Type: text/html; charset=utf-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< Set-Cookie: __cfduid=███████████████████████████████████████████; expires=Tue, 09-Jan-18 19:38:36 GMT; path=/; domain=.externals.io; HttpOnly
< Set-Cookie: PHPSESSID=██████████████████████████; path=/
< Vary: Accept-Encoding
< Server: cloudflare-nginx
< CF-RAY: ████████████████-███
< 
```
It happens even if the page stays opened in the browser, so it looks like the session data gets garbage collected on the server side. I didn't explore the source code much.

Is it possible to store authentication-specific data in a long living cookie and not rely on PHP session?

**Comment by mnapoli** [comment]
Hi, agreed, I have to log back again every time and it's a bit annoying. Currently indeed the logged in user is stored in the session: https://github.com/mnapoli/externals/blob/master/src/Application/Controller/AuthController.php#L89

I'm OK to store data in cookies but it has to be safe (stealing cookies should not leak data, and cookies should not be forgeable). I don't have time to look into a solution at the moment but suggestions and pull requests are welcome. I can also increase session time on the server, but I don't know if that's the best course of action.

**Comment by morozov** [comment]
@mnapoli what about using [JWT](https://jwt.io/) stored in a cookie? I could try putting together an example if you're okay with the approach?

**Comment by mnapoli** [comment]
Good idea, there's https://github.com/psr7-sessions/storageless as a PSR-7 middleware too so maybe we can reuse that?

**Comment by morozov** [comment]
@mnapoli the package you suggested above works well at a glance. The only issue I'm experiencing is that it [enforces secure cookies](https://github.com/psr7-sessions/storageless/blob/2.0.0/src/PSR7Session/Http/SessionMiddleware.php#L132) by default and it's not configurable. Given that [externals.io](http://externals.io/) works in both modes (and I only figured out HTTPS is available after seeing this issue), the replacement session mechanism will not work over HTTP.

Are you considering adding enforced HTTP→HTTPS redirect to your server configuration so that we can make sure the new cookies are always accepted?

**Comment by mnapoli** [comment]
Implemented in #20!
